packages/magento-product/components/ProductPageMeta/ProductPageMeta.tsx

```diff
@@ -3,14 +3,25 @@ import { useProductLink } from '../../hooks/useProductLink'
 import { ProductPageMetaFragment } from './ProductPageMeta.gql'
 
 export function ProductPageMeta(props: ProductPageMetaFragment) {
-  const { name, meta_title, meta_description, url_key, __typename } = props
+  const { sku, categories, price_range, media_gallery, name, meta_title, meta_description, url_key, __typename } = props
   const productLink = useProductLink({ url_key, __typename })
 
   return (
     <PageMeta
-      title={meta_title ?? name ?? ''}
-      metaDescription={meta_description ?? name ?? ''}
-      canonical={productLink}
-    />
+        title={meta_title ?? name ?? ''}
+        metaDescription={meta_description ?? name ?? ''}
+        canonical={productLink}
+    >
+      {sku && <>
+          <meta property='type' content='product' key='og-type' />
+          <meta property='product:retailer_part_no' content={sku} key='og-sku' />
+      </>}
+      {name && <meta property='og:title' content={name} key='og-title' />}
+      {media_gallery?.[0]?.url && <meta property='og:image' content={media_gallery[0].url} key='og-image' />}
+      {price_range?.maximum_price?.regular_price?.value && <meta property='product:price:amount' content={price_range.maximum_price.regular_price.value.toString()} key='og-price' />}
+      {price_range?.maximum_price?.final_price?.value && <meta property='product:sale_price:amount' content={price_range.maximum_price.final_price.value.toString()} key='og-sale-price' />}
+      {categories && categories.map((category) => category?.name ? <meta property='product:category' content={category.name} key={category.uid} /> : null)}
+      {productLink && <meta property='og:url' content={productLink} key='og-url' />}
```

Comment:
The productLink isn't a fully qualified URL, this should be the full canonical URL. Maybe we can move this next to the canonical?

Reply:
You're right, I've converted it here so that all the og:tags would be defined in the same place, let me know if you think it's better to use the canonical that we already have in the next-ui PageMeta and set it also to the og:url there.